crates/vm/src/system/program/tests/mod.rs

```diff
@@ -47,29 +46,34 @@ fn interaction_test(program: Program<BabyBear>, execution: Vec<u32>) {
 
     let counter_air = DummyInteractionAir::new(9, true, bus.0);
     let mut program_cells = vec![];
-    for (pc_idx, instruction) in instructions.iter().enumerate() {
-        program_cells.extend(vec![
-            BabyBear::from_canonical_usize(execution_frequencies[pc_idx]), // hacky: we should switch execution_frequencies into hashmap
-            BabyBear::from_canonical_usize(pc_idx * (DEFAULT_PC_STEP as usize)),
-            instruction.opcode.to_field(),
-            instruction.a,
-            instruction.b,
-            instruction.c,
-            instruction.d,
-            instruction.e,
-            instruction.f,
-            instruction.g,
-        ]);
+    for (index, frequency) in execution_frequencies.into_iter().enumerate() {
+        let option = program.get_instruction_and_debug_info(index);
+        if option.is_some() {
+            let instruction = option.unwrap().0;
+            program_cells.extend(vec![
+                BabyBear::from_canonical_usize(frequency), // hacky: we should switch execution_frequencies into hashmap
```

Comment:
on the hackiness, do you mean just in test, or in general? the goal of using the gaps was so execution_frequencies could stay a lookup instead of a hashmap